code/game/objects/items/tools.dm

```diff
@@ -463,12 +463,31 @@
 		if(length(container.reagents.reagent_list) > 1)
 			to_chat(user, "<span class='warning'>[container] has too many chemicals mixed into it. You wouldn't want to put the wrong chemicals into [src].</span>")
 			return ..()
-		if(amountNeeded > 0 && container.reagents.has_reagent("welding_fuel"))
+		if(amountNeeded <= 0)
+			to_chat(user, "<span class='warning'>[src] is already full.</span>")
+			return ..()
+		if(container.reagents.has_reagent("welding_fuel"))
 			container.reagents.trans_id_to(src, "welding_fuel", amountNeeded)
 			to_chat(user, "<span class='notice'>You transfer some fuel from [container] to [src].</span>")
+			return ..()
+		if(container.reagents.has_reagent("plasma"))
+			container.reagents.trans_id_to(src, "plasma", amountNeeded)
+			to_chat(user, "<span class='notice'>You slip some plasma from [container] to [src].</span>")
+			if(!reagents.has_reagent("plasma"))
+				message_admins("[key_name_admin(user)] rigged a welder with plasma.")

```

Comment:
there's really no need to spam admins about things like this all the time

99% of cases of this should be absolutely fine

Reply:
If its not going to alert admins it should atleast be logged.

Reply:
a contributors default tool to reach for should be logging, not admin alerts

Reply:
You're right, what's the right way to log instead of alert?

Reply:
https://github.com/tgstation/tgstation/blob/master/code/__HELPERS/_logging.dm#L46

Reply:
https://github.com/tgstation/tgstation/search?utf8=%E2%9C%93&q=log_game&type=